packages/preview/src/browser/preview-widget.ts

```diff
@@ -135,14 +135,20 @@ export class PreviewWidget extends BaseWidget {
         this.performUpdate();
     }
 
+    protected lastContentHash: number = 0;
     protected async performUpdate(): Promise<void> {
         if (!this.resource) {
             return;
         }
         const uri = this.resource.uri;
         const document = this.workspace.textDocuments.find(d => d.uri === uri.toString());
-        const content: MaybePromise<string> = document ? document.getText() : this.resource.readContents();
-        const contentElement = await this.render(await content, uri);
+        const content: MaybePromise<string> = document ? document.getText() : await this.resource.readContents();
+        const contentHash = this.hashCode(content);
+        if (contentHash === this.lastContentHash) {
```

Comment:
hash can be the same for different content, is not it? why don't compare content itself?

Reply:
👍

Reply:
revised to compare content.